### PR TITLE
Fix lenovo-thinklmi and add a new case for it

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -6937,6 +6937,13 @@ fu_engine_check_firmware_attributes(FuEngine *self, FuDevice *device)
 	subsystem = fu_udev_device_get_subsystem(FU_UDEV_DEVICE(device));
 	if (g_strcmp0(subsystem, "firmware-attributes") == 0) {
 		g_autoptr(GError) error = NULL;
+		if (add) {
+			FuBiosSettings *settings = fu_context_get_bios_settings(self->ctx);
+			if (fu_bios_settings_get_count(settings) > 0) {
```
Comment (hughsie) [inline src/fu-engine.c:6942]: this seems like a single-use API call in the slow path -- can't we get the entire array and just check the length ourselves?

File: src/fu-engine.c [hunk 2]
```diff
@@ -6926,7 +6926,7 @@ fu_engine_apply_default_bios_settings_policy(FuEngine *self, GError **error)
 }
 
 static void
-fu_engine_check_firmware_attributes(FuEngine *self, FuDevice *device)
+fu_engine_check_firmware_attributes(FuEngine *self, FuDevice *device, gboolean add)
```
Comment (hughsie) [inline src/fu-engine.c:6929]: maybe `add` to `added` just to make it a bit clearer

File: libfwupdplugin/fu-context.c [hunk 3]
```diff
@@ -590,6 +590,7 @@ fu_context_load_hwinfo(FuContext *self, GError **error)
 	GPtrArray *guids;
 	g_autoptr(GError) error_smbios = NULL;
 	g_autoptr(GError) error_hwids = NULL;
+	g_autoptr(GError) error_fw_attributes = NULL;
```
Comment (hughsie) [inline libfwupdplugin/fu-context.c:593]: fw_settings, surely :)

File: plugins/lenovo-thinklmi/fu-plugin-lenovo-thinklmi.c [hunk 4]
```diff
@@ -27,14 +28,29 @@ fu_plugin_lenovo_thinklmi_startup(FuPlugin *plugin, FuProgress *progress, GError
 }
 
 static void
-fu_plugin_lenovo_thinklmi_device_registered(FuPlugin *plugin, FuDevice *device)
+fu_plugin_lenovo_thinklmi_cpu_registered(FuPlugin *plugin, FuDevice *device)
+{
+	/* Ryzen 6000 doesn't support S3 even if the BIOS offers it */
+	if (fu_device_has_guid(device, "CPUID\\PRO_0&FAM_19&MOD_44")) {
```
Comment (hughsie) [inline plugins/lenovo-thinklmi/fu-plugin-lenovo-thinklmi.c:34]: has_instance_id, no?

File: plugins/lenovo-thinklmi/fu-plugin-lenovo-thinklmi.c [hunk 5]
```diff
@@ -55,6 +66,16 @@ fu_plugin_lenovo_thinklmi_device_registered(FuPlugin *plugin, FuDevice *device)
...
 }
 
+static void
+fu_plugin_lenovo_thinklmi_device_registered(FuPlugin *plugin, FuDevice *device)
+{
+	if (g_strcmp0(fu_device_get_plugin(device), "uefi_capsule") == 0)
+		fu_plugin_lenovo_thinklmi_uefi_capsule_registered(fu_plugin_get_context(plugin),
+								  device);
+	else if (g_strcmp0(fu_device_get_plugin(device), "cpu") == 0)
+		fu_plugin_lenovo_thinklmi_cpu_registered(fu_plugin_get_context(plugin), device);
```
Comment (hughsie) [inline plugins/lenovo-thinklmi/fu-plugin-lenovo-thinklmi.c:76]: braces here perhaps

File: src/fu-engine.c (hunk 2 above)
Comment (hughsie) [reply src/fu-engine.c:6929]: Added too?

**Review by hughsie — APPROVED** [review]

**Comment by superm1** [comment]
@mrhpearson FYI on this MR.  (I'm pretty sure this is a BIOS bug you'll want to raise that the setting is exporting anyway)

**Comment by hughsie** [comment]
Yup, I was going to say too; we don't want fwupd papering over this kind of thing forever.

**Comment by mrhpearson** [comment]
Just as a note - BIOS team have done the fix so you can't adjust sleep setting using WMI. It should be in the next release

**Comment by superm1** [comment]
> Just as a note - BIOS team have done the fix so you can't adjust sleep setting using WMI. It should be in the next release

It's probably an infinitesimally small audience, but do they move the sleep setting out of S3 if someone happened to have set it to this using the WMI interface but upgrades to the new BIOS?

We mitigate it from fwupd due to this PR, but not to say someone couldn't do it from Windows over WMI or Linux directly in the kernel.

**Comment by mrhpearson** [comment]
No idea - I can't imagine anybody did it but if they did...I guess they'd need to go back to the previous BIOS and fix it). 
I'll check and see what they say.

**Pull request merged**